src/eigen/diag.jl

```diff
@@ -45,7 +45,8 @@ function diagonalize_all_kblocks(eigensolver, ham::Hamiltonian, nev_per_kpoint::
 
         prec = nothing
         prec_type !== nothing && (prec = prec_type(ham.basis, kpt))
-        results[ik] = eigensolver(ham.blocks[ik], guessk;
+        results[ik] = eigensolver(ham.blocks[ik],
+                                  copy(guessk);  # Copy in case eigensolver modifies guess
```

Comment:
Not super happy about this copy... Can't we just enforce that the eigensolver doesn't modify?

Reply:
We probably should. That was a quick fix I needed for something else. I'll try to see if I can fix it at another level.